src/octree/batch_iterator.rs

```diff
@@ -185,10 +205,33 @@ impl<'a> BatchIterator<'a> {
                         ))
                         .into()),
                     };
-                    let point_iterator = octree.points_in_node(point_location, node_id);
+
+                    // one pointstream per thread vs one per node allows to send more full point batches
                     let mut point_stream =
                         PointStream::new(batch_size, local_from_global, &send_func);
-                    let _ = point_stream.push_points_and_callback(point_iterator);
+
+                    loop {
```

Comment:
How about replacing this by a more concise version such as
```rust
                    while let Some((node_id, octree)) = worker.pop().or_else(|| {
                        std::iter::repeat_with(|| jobs.steal_batch_and_pop(&worker))
                            .find(|task| !task.is_retry())
                            .and_then(|task| task.success())
                    }) {
                        let point_iterator = octree.points_in_node(&point_location, node_id);
                        //executing on the available next task if the fuction still requires it
                        match point_stream.push_points_and_callback(point_iterator) {
                            Ok(_) => continue,
                            Err(ref e) => match e.kind() {
                                ErrorKind::Channel(ref _s) => break, // done with the function computation
                                _ => panic!("BatchIterator: Thread error {}", e), //some other error
                            },
                        }
                    }
```
?

Reply:
done!